projects/storefrontlib/cms-components/product/carousel/product-carousel/product-carousel.component.ts

```diff
@@ -52,6 +55,32 @@ export class ProductCarouselComponent {
    */
   items$: Observable<Observable<Product | undefined>[]> =
     this.componentData$.pipe(
+      switchMap((data) => {
+        const categoryCodes = data?.categoryCodes?.split(' ');
```

Comment:
Because this part of the observable is just for joining the product categories, could we move it to its own function just to make it easier to understand what is happening at a glance? Something like `switchMap(data => joinProductCodes)`

Reply:
fixed